R/oc_bbox.R

```diff
@@ -91,11 +91,15 @@ oc_bbox.default <- function(xmin, ymin, xmax, ymax, ...) { # nolint - see lintr
 #' @name oc_bbox
 #' @export
 oc_bbox.data.frame <- function(data, xmin, ymin, xmax, ymax, ...) { # nolint - see lintr issue #223
-  xmin <- data[[deparse(substitute(xmin))]]
-  ymin <- data[[deparse(substitute(ymin))]]
-  xmax <- data[[deparse(substitute(xmax))]]
-  ymax <- data[[deparse(substitute(ymax))]]
-  oc_bbox(xmin, ymin, xmax, ymax)
+  xmin <- rlang::enquo(xmin)
+  ymin <- rlang::enquo(ymin)
+  xmax <- rlang::enquo(xmax)
+  ymax <- rlang::enquo(ymax)
+
+  oc_bbox(xmin = rlang::eval_tidy(xmin, data = data),
```

Comment:
Still trying to wrap my head around it too. Basically, I think the difference is between evaluation with `eval_tidy()` and unquoting with `!!`. In this instance, we want "xmin" etc. to be evaluated so that it is the numeric vector from the data frame. On the other hand, `!!` unquotes within a function call, especially within `dplyr` functions that use this quasiquotation. `bind_cols = FALSE` in `oc_forward_df()` and `oc_reverse_df()` uses `eval_tidy()`, because we want access to the underlying vectors to be used in `oc_forward()`. On the other hand, `bind_cols = TRUE` uses `!!`, because it is being called within `dplyr::mutate()`.